controllers/fenceagentsremediation_controller_test.go

```diff
@@ -0,0 +1,214 @@
+			Expect(k8sClient.Delete(context.Background(), templateTest)).NotTo(HaveOccurred())
+			Expect(k8sClient.Delete(context.Background(), test)).NotTo(HaveOccurred())
+		})
+
+		When("creating FAR CR", func() {
+			It("should build the exec command based on Template and Remediation CRs", func() {
+				Eventually(func() bool {
+					return cliCommandsEquality(templateTest, test)
+				}, 1*time.Second, 500*time.Millisecond).Should(BeTrue())
+			})
+		})
+	})
+})
+
+
+// buildFARTemplate from string to arrays to two string maps (key-value manner)
+func buildFARTemplate(fields []string, values []string, nodeFields []string, nodeValues []string, node string) (map[v1alpha1.ParameterName]string, map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string) {
+	testShareParam := make(map[v1alpha1.ParameterName]string)
+	testNodeParam := make(map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string)
+	i := 0
+	for i = 0; i < len(values); i++ {
+		field := v1alpha1.ParameterName(fields[i])
+		testShareParam[field] = values[i]
+	}
+
+	nodeName := v1alpha1.NodeName(node)
+	numNodeParam := len(fields) - len(values)
+	for j := 0; j < numNodeParam; j++ {
+		if indexOf(node, nodeFields) > -1 {
+			field := v1alpha1.ParameterName(fields[i+j])
+			testNodeParam[field] = make(map[v1alpha1.NodeName]string)
+			testNodeParam[field][nodeName] = nodeValues[indexOf(node, nodeFields)]
+		}
+	}
+	return testShareParam, testNodeParam
+}
+
+// indexOf return the index of element in data array. If it is not found, return -1
+func indexOf(element string, data []string) int {
+	for k, v := range data {
+		if element == v {
+			return k
+		}
+	}
+	return -1 
+}
+
+// newFenceAgentsRemediationTemplate assign the input to the FenceAgentsRemediationTemplate's Spec
+func newFenceAgentsRemediationTemplate(agent string, sharedparameters map[v1alpha1.ParameterName]string, nodeparameters map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string) *v1alpha1.FenceAgentsRemediationTemplate {
+	return &v1alpha1.FenceAgentsRemediationTemplate{
+		ObjectMeta: metav1.ObjectMeta{Name: fenceAgentsTemplateName, Namespace: defaultNamespace},
+		Spec: v1alpha1.FenceAgentsRemediationTemplateSpec{
+			Agent:            agent,
+			SharedParameters: sharedparameters,
+			NodeParameters:   nodeparameters,
+		},
+	}
+}
+
+// buildFarPod builds a dummy pod with FAR label and namespace
+func buildFarPod() * corev1.Pod {
+	fenceAgentsPod := &corev1.Pod{}
+	fenceAgentsPod.Labels = faPodLabels
+	fenceAgentsPod.Name = "mock-fence-agents"
+	fenceAgentsPod.Namespace = defaultNamespace
+	container := corev1.Container{
+		Name:  "foo",
+		Image: "foo",
+	}
+	fenceAgentsPod.Spec.Containers = []corev1.Container{container}
+	return fenceAgentsPod
+}
+
+// cliCommandsEquality creates the command for CLI and compares it with the production command
+func cliCommandsEquality(farTemplate *v1alpha1.FenceAgentsRemediationTemplate, far *v1alpha1.FenceAgentsRemediation) bool {
```

Comment:
Since this method is already called from an `Eventually` block, I suggest removing inner `Expect` verification - just return `false` instead